File: frontend/src/types/game.ts
export type Player = 'X' | 'O';

export type GameStatus = 'IN_PROGRESS' | 'WON' | 'DRAW';

export interface GameBoard {
  cells: (Player | null)[][];
}

export interface Move {
  player: Player;
  position: number;
}

export interface GameState {
  board: GameBoard;
  currentPlayer: Player;
  status: GameStatus;
  winner: Player | 'DRAW' | null;
  moveCount: number;
  moveHistory: Move[];
}

File: backend/game-session-service/src/main/java/com/edu/tictactoe/gamesession/domain/Game.java
package com.edu.tictactoe.gamesession.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Game(
    String id,
    String status,
    String winner,
    @JsonProperty("currentPlayer") String currentPlayer,
    @JsonProperty("moveCount") int moveCount,
    @JsonProperty("board") Board board
) {
    public boolean isGameOver() {
        return "WON".equals(status) || "DRAW".equals(status);
    }
    
    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Board(
        List<Cell> cells
    ) {}
    
    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Cell(
        boolean empty
    ) {}
}
